examples/powercli/nsx-t_tag-sync/handler/script.ps1

```diff
@@ -0,0 +1,160 @@
+# Author: Craig Straka (craig.straka@it-partners.com)
+# Version .1 (Beta)
+# Master stored at: https://github.com/cstraka/NSX-T_Tag-Sync (/nsx)
+# Intention:
+#   Synchronize vSphere tags with NSX-T tags unidiretionally (vSphere is the master)
+#   Script is used as a OpenFaas VMware Event Broker powercli function to intercept and parse vSphere events from event types:
+#       - com.vmware.cis.tagging.attach
+#       - com.vmware.cis.tagging.detach
+#   Script gets vSphere Tags from vSphere, transforms the data, and updates the Virtual machine object in NSX-T.
+# 
+# Tested with:
+#   VEBA .5 (OpenFaaS)
+#   vSphere 7.0.1
+#   NSX-T 3.0.2
+#
+# Notes:
+#   Machine names in a vCenter instance must be unique: 
+#       Script has no way, based on limited specificity of vSphere event message of the types above, to discern correct machine other than by name.
+#           Lots of issues here as the name may not be unique in the cluster
+#           Intention is to fix this as vSphere event messages evolve.
+#       Event data has a number of odd charachters, such as new lines, that must be accomodated to get the machine name.  
+#           accomodation efforts to handle naming are ongoing as errata is reported.
+#
+# Planned enhancements (script today is minimally viable):
+#   Updates based on vSphere event changes
+#   Hopefully, better vSphere and NSX-T integration will render this script obsolete.
+
+# Process function Secrets passed in
+$SECRETS_FILE = "/var/openfaas/secrets/nsx-secrets"
+$SECRETS_CONFIG = (Get-Content -Raw -Path $SECRETS_FILE | ConvertFrom-Json)
+
+# Test if PowerCLI Module is installed, install if not
+if(Get-Module -ListAvailable -Name VMware.VimAutomation.Core) {
+    Write-Host "Module exists"
+} else {
+    Write-Host "Module does not exist"
+    Install-Package -Name VMware.VimAutomation.Core
+}
```

Comment:
If you use our base PowerCLI container image, then PowerCLI will already be installed. You definitely want this baked in versus having to check and then halt function execution to install module if you were building your own image

Reply:
I tried a few images and always had the same problem... VMware.VimAutomation.Core wasn't loaded and so had to install the first time it ran.  The container image I used is based on vmware/powerclicore... Is there another I should use?  Regardless, I'll give it a go again.

Thanks for the feedback!